inference-engine/src/plaidml_plugin/ops/extract_image_patches.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset1.hpp"
+#include "ngraph/opsets/opset3.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;         // NOLINT[build/namespaces]
+using namespace InferenceEngine; // NOLINT[build/namespaces]
+
+namespace {
+
+template <typename T>
+edsl::Tensor createKernelTensor(std::vector<int64_t> &filterShape,
+                                edsl::Tensor inputTensor) {
+  auto inputType = inputTensor.dtype();
+  auto inputShape = inputTensor.compute_shape().sizes();
+  auto inputChannel = inputShape[1];
+  // build kernel Tensor dims.
+  auto depths = filterShape[0] * filterShape[1] * inputChannel;
+  std::vector<int64_t> kernelDims{
+      /*output channels*/ depths,
+      /*input channels*/ inputChannel,
+      /*filter width*/ filterShape[0],
+      /*filter height*/ filterShape[1],
+  };
+  // kernel tensor element size.
+  size_t kernelSum = 1;
+  for (auto dim : kernelDims) {
+    kernelSum *= dim;
+  }
+  // build one-zero kernel Tensor.
+  std::vector<T> data(kernelSum, 0);
+
+  int64_t channel = 0;
+  for (int64_t depth = 0; depth < depths; depth++) {
+    auto index = depth * kernelDims[1] * kernelDims[2] * kernelDims[3] +
+                 channel * kernelDims[2] * kernelDims[3] +
+                 (depth / inputChannel) / filterShape[1] * kernelDims[3] +
+                 (depth / inputChannel) % filterShape[1];
+    data[index] = 1;
+    if (++channel == inputChannel) {
+      channel = 0;
+    }
+  }
+  TensorShape shape(inputType, kernelDims);
+  Buffer buffer(shape);
+  buffer.copy_from(data.data());
+
+  return edsl::cast(edsl::Constant(buffer, "Kernel"), inputType);
```

Comment:
Is this cast needed?

Reply:
if we only support FP32 and INT32, we don't need this cast, after testing, it all work fine. I will remove this.